src/storm/transformer/AddUncertainty.cpp

```diff
@@ -12,24 +16,53 @@ AddUncertainty<ValueType>::AddUncertainty(std::shared_ptr<storm::models::sparse:
 
 template<typename ValueType>
 std::shared_ptr<storm::models::sparse::Model<Interval>> AddUncertainty<ValueType>::transform(double additiveUncertainty, double minimalTransitionProbability) {
-    storage::SparseMatrixBuilder<storm::Interval> newMatrixBuilder =
-        storage::SparseMatrixBuilder<storm::Interval>(origModel->getTransitionMatrix().getRowCount(), origModel->getTransitionMatrix().getColumnCount());
+    // we first build the matrix and later copy the row grouping.
+    auto newMatrixBuilder =
+        storage::SparseMatrixBuilder<storm::Interval>(origModel->getTransitionMatrix().getRowCount(), origModel->getTransitionMatrix().getColumnCount(),
+                                                      origModel->getTransitionMatrix().getNonzeroEntryCount(), true, false);
+    // Build transition matrix (without row grouping)
     for (uint64_t state = 0; state < origModel->getNumberOfStates(); ++state) {
-        uint64_t endOfRowGroup = origModel->getTransitionMatrix().getRowGroupIndices()[state + 1];
-        for (uint64_t row = origModel->getTransitionMatrix().getRowGroupIndices()[state]; row < endOfRowGroup; ++row) {
+        for (auto row : origModel->getTransitionMatrix().getRowGroupIndices(state)) {
             for (auto const& entry : origModel->getTransitionMatrix().getRow(row)) {
                 newMatrixBuilder.addNextValue(row, entry.getColumn(), addUncertainty(entry.getValue(), additiveUncertainty, minimalTransitionProbability));
             }
         }
     }
 
+    storm::storage::sparse::ModelComponents<storm::Interval> modelComponents(newMatrixBuilder.build(), origModel->getStateLabeling());
+    if (!origModel->getTransitionMatrix().hasTrivialRowGrouping()) {
+        modelComponents.transitionMatrix.setRowGroupIndices(origModel->getTransitionMatrix().getRowGroupIndices());
+    }
+    // Change value type of standard reward model.
+    std::unordered_map<std::string, models::sparse::StandardRewardModel<storm::Interval>> newRewardModels;
+    for (auto const& rewModel : origModel->getRewardModels()) {
+        auto const& oldRewModel = rewModel.second;
+        std::optional<std::vector<storm::Interval>> stateRewards;
+        std::optional<std::vector<storm::Interval>> stateActionRewards;
+        if (oldRewModel.hasStateRewards()) {
+            stateRewards = utility::vector::convertNumericVector<storm::Interval>(oldRewModel.getStateRewardVector());
+        }
+        if (oldRewModel.hasStateActionRewards()) {
+            stateActionRewards = utility::vector::convertNumericVector<storm::Interval>(oldRewModel.getStateRewardVector());
+        }
+        STORM_LOG_THROW(oldRewModel.hasTransitionRewards(), exceptions::NotImplementedException, "Transition rewards are not supported.");
```

Comment:
```suggestion
        STORM_LOG_THROW(!oldRewModel.hasTransitionRewards(), exceptions::NotImplementedException, "Transition rewards are not supported.");
```

Reply:
Why dont my tests currently fail... I find this very confusing :-)